### ISSUE TITLE
Add CoAtNet Models 

### ISSUE THREAD

**Issue by IMvision12** [keras-team/keras-cv#1384]
CoAtNet: Marrying Convolution and Attention for All Data Sizes
research Paper : [paper](https://arxiv.org/abs/2106.04803)

Contnet is a network which is combination of depthwise Convolution and self-Attention, stacking convolution layers and attention layers vertically. CoAtNet achieves 86.0% ImageNet top-1 accuracy; When pre-trained with 13M images from ImageNet-21K, CoAtNet achieves 88.56% top-1 accuracy.
![210107373-eda24426-148c-4c37-ae9b-aac96eb2809b](https://user-images.githubusercontent.com/88665786/217829579-1d4cf5b1-4632-4b9b-a3be-f7803ea0d8d8.png)

As per Image Classification on ImageNet : https://paperswithcode.com/sota/image-classification-on-imagenet

CoatNet-7 is ranked fifth and has a top-1 accuracy of 90.88% on ImageNet.

Will it be worthwhile to include this model? if yes I Would like to work on this!!!
@LukeWood @ianstenbit

**Comment by ianstenbit** [comment]
Hey @IMvision12 -- right now, new backbones are not a high priority. We'd be open to this in the future, but for now we're going to pause accepting new backbones.

Our highest priority task right now (which is open to contributors) is improving performance of preprocessing layers (see #1373). 

There are a number of open issues about this if you'd be interested (e.g. #1376). Thanks!

**Comment by IMvision12** [comment]
@ianstenbit Okay, let me know when we resume adding backbones and models, coz I am interested in adding many models to kerasCV like swin-transformers, coatnet, etc and some segmentation models likes Segformer, MaskFormer etc!!

Thanks

**Comment by ianstenbit** [comment]
Right now we're not looking for new backbones unless they are critical for a new high-level model (e.g. MiT for SegFormer)
